include/cpp-statsd-client/StatsdClient.hpp

```diff
@@ -291,35 +286,36 @@ inline void StatsdClient::send(const std::string& key,
     std::stringstream valueStream;
     valueStream << std::fixed << std::setprecision(m_gaugePrecision) << value;
 
-    m_buffer.clear();
+    std::string buffer;
+    buffer.reserve(256);
```

Comment:
i still cant believe this would be faster than keeping the buffer around! allocating once in the constructor and then using clear (which is O(1)) should always be faster than this.

maybe we could just make a static thread local buffer to achieve the same result and avoid the allocations/deallocations? something like:

```suggestion
    // the thread keeps this buffer around and reuses it, clear should be O(1) and reserve should only have to do so the first time, after that its a no-op
    static thread_local std::string buffer;
    buffer.clear();
    buffer.reserve(256);
```

Reply:
It isn't though. Single threaded performance of allocating once in the constructor is 3673.82 ms, whereas local allocation is 4002.7 ms. And we can't trust the multi-threaded performance of allocating in the constructor because it does the wrong thing.

Reply:
> And we can't trust the multi-threaded performance of allocating in the constructor because it does the wrong thing.

Yeah, I wasn't concerned about multithreaded perf because testing thread-unsafe code with threads makes no sense :smile:

Reply:
... And this is the constructor allocation performance from today.
```
user@group:~$ g++ -o statsd_example statsd_example.cpp -I./cpp-statsd-client/ -lpthread && ./statsd_example 1 1000000
Total metrics sent: 1000000
Time taken: 2780.39 ms
Metrics sent per second: 359662
```